sharing/src/test/scala/io/delta/sharing/spark/DeltaFormatSharingSourceSuite.scala

```diff
@@ -0,0 +1,866 @@
+          processAllAvailableInStream(
+            Map(
+              "responseFormat" -> "delta",
+              "startingVersion" -> "0",
+              SKIP_CHANGE_COMMITS_OPTION -> "true"
+            ),
+            CheckAnswer("keep1", "keep2")
+          )
+          assertBlocksAreCleanedUp()
+        }
+      }
+    }
+  }
+
+  test("streaming works with DV") {
+    withTempDir { inputDir =>
+      val deltaTableName = "delta_table_dv"
+      withTable(deltaTableName) {
+        createSimpleTable(deltaTableName, enableCdf = false)
+        spark.sql(
+          s"ALTER TABLE $deltaTableName SET TBLPROPERTIES('delta.enableDeletionVectors' = true)"
+        )
+        val sharedTableName = "shared_streaming_table_dv"
+        prepareMockedClientMetadata(deltaTableName, sharedTableName)
+        val profileFile = prepareProfileFile(inputDir)
+        val tablePath = profileFile.getCanonicalPath + s"#share1.default.$sharedTableName"
+
+        withSQLConf(getDeltaSharingClassesSQLConf.toSeq: _*) {
+          def InsertToDeltaTable(values: String): Unit = {
+            sql(s"INSERT INTO $deltaTableName VALUES $values")
+          }
+
+          def processAllAvailableInStream(
+              sourceOptions: Map[String, String],
+              expectations: StreamAction*): Unit = {
+            val df = spark.readStream
+              .format("deltaSharing")
+              .options(sourceOptions)
+              .load(tablePath)
+              .filter($"c2" contains "keep")
+              .select("c1")
+
+            val base = Seq(StartStream(), ProcessAllAvailable())
+            testStream(df)((base ++ expectations): _*)
+          }
+
+          // Insert at version 2.
+          InsertToDeltaTable("""(1, "keep1"),(2, "keep1"),(3, "keep1"),(1,"drop1")""")
+          // delete at version 3.
+          sql(s"""DELETE FROM $deltaTableName WHERE c1 >= 2 """)
+
+          prepareMockedClientAndFileSystemResult(
+            deltaTable = deltaTableName,
+            sharedTable = sharedTableName,
+            versionAsOf = Some(3L)
+          )
+          prepareMockedClientGetTableVersion(deltaTableName, sharedTableName)
+          processAllAvailableInStream(
+            Map("responseFormat" -> "delta"),
+            CheckAnswer(1)
+          )
+
+          prepareMockedClientAndFileSystemResultForStreaming(
+            deltaTableName,
+            sharedTableName,
+            startingVersion = 0,
+            endingVersion = 3,
+            assertDVExists = true
+          )
+
+          // The streaming query will fail because deletes detected in version 3. And there are no
+          // options provided to ignore the deletion.
+          val e = intercept[Exception] {
+            processAllAvailableInStream(
+              Map("responseFormat" -> "delta", "startingVersion" -> "0")
+            )
+          }
+          for (msg <- Seq(
+              "Detected a data update",
```

Comment:
nit: why can't you just use the wild card within the expected string for verifying just once? Seems like this type of check is adding decent amount of lines to the test code.

Reply:
will check